engine/resources/exec_test.go

```diff
@@ -257,3 +260,62 @@ func TestExecTimeoutBehaviour(t *testing.T) {
 
 	// no error
 }
+
+func TestExecAutoEdge1(t *testing.T) {
+
+	g, err := pgraph.NewGraph("TestGraph")
+	if err != nil {
+		t.Errorf("error creating graph: %v", err)
+		return
+	}
+
+	r1 := &ExecRes{
+		Base: traits.Base{
+			Kinded: traits.Kinded{Xkind: "exec"},
+			Named:  traits.Named{Xname: "true"},
+		},
+		Cmd:   "/somefile",
+		User:  "someuser",
+		Group: "somegroup",
+	}
+	r2 := &UserRes{
+		Base: traits.Base{
+			Kinded: traits.Kinded{Xkind: "user"},
+			Named:  traits.Named{Xname: "someuser"},
+		},
+	}
+	r3 := &GroupRes{
+		Base: traits.Base{
+			Kinded: traits.Kinded{Xkind: "group"},
+			Named:  traits.Named{Xname: "somegroup"},
+		},
+	}
+	r4 := &FileRes{
+		Base: traits.Base{
+			Kinded: traits.Kinded{Xkind: "file"},
+			Named:  traits.Named{Xname: "/somefile"},
+		},
+	}
+	g.AddVertex(r1, r2, r3, r4)
+
+	if i := g.NumEdges(); i != 0 {
+		t.Errorf("should have 0 edges instead of: %d", i)
+		return
+	}
+
+	debug := testing.Verbose() // set via the -test.v flag to `go test`
+	logf := func(format string, v ...interface{}) {
+		t.Logf("test: "+format, v...)
+	}
+	// run artificially without the entire engine
+	if err := autoedge.AutoEdge(g, debug, logf); err != nil {
+		t.Errorf("error running autoedges: %v", err)
+		return
+	}
+
+	// two edges should have been added
+	if i := g.NumEdges(); i != 3 {
+		t.Errorf("should have 3 edges instead of: %d", i)
+		return
+	}
```

Comment:
Do you think you might be convinced to add some additional easy logic to check (1) that these are the edges we actually expect, and (2) that they're pointing in the correct directions? You should be able to see the pointers match what we expect. If you don't know how, I can write this addendum to the patch. Cheers